utils.py

```diff
@@ -345,25 +348,36 @@ def get_runmode():
         return 'script'
 
 
-def query_packages(packages):
+def query_packages(mode, packages):
     if config.SKIP_DEPENDENCIES:
         return
 
     missing_packages = []
+    bad_packages = []
 
     for p in packages:
         command = f"{config.PACKAGE_MANAGER_COMMAND_QUERY}{p}"
         logging.debug(f"pkg query command: \"{command}\"")
         result = subprocess.run(command, stdout=subprocess.PIPE, stderr=subprocess.PIPE, shell=True, text=True)
         logging.debug(f"pkg query result: {result.returncode}")
-        if result.returncode != 0:
+        if result.returncode != 0 and mode == "install":
             missing_packages.append(p)
```

Comment:
You can simplify this function by setting a default value for mode:
```python
def query_packages(packages, mode='install'):
```
This would remove the `elif` code on L367-368 and make passing `mode` optional when calling it.